src/lib/utils.c

```diff
@@ -483,7 +485,9 @@ int match_string(const char *string, const char *pattern)
 
 	status = regcomp(&regex, pattern, REG_EXTENDED);
 	if (status != 0) {
-		printf("regecomp failed, ret=%d", status);
```

Comment:
I see this as badly implemented functionality.  If pattern matches by `strcmp` we are exiting. If not we are running regcomp anyway- I didn't test how this code will behave when not regular expression is passed but so fair I know nobody reported issue so I think it is successful.

For example, we have a few controllers, let say:
```
/sys/devices/ahci
/sys/devices/scsi
/sys/devices/vmd
```
and we put:
`ALLOWLIST=/sys/devices/vmd` in conf. Then it exites early for `vmd` but regex it executed for other types (and error is printed? - I'm not sure here)

Maybe we should keep regex_t on white or black list and compile regex during conf file parsing. That will give us possibility to stop application if regex if wrong. Then we will no need to report error here.

@tasleson I know that all you wanted is to fix the issue but I would like to address issue I discovered. Could you please test it and fix?

Reply:
> @tasleson I know that all you wanted is to fix the issue but I would like to address issue I discovered. Could you please test it and fix?

Yes, all I wanted to do was prevent us from outputting to `stdout` in the library which is bad.  I would prefer we integrate this change and open an issue to address the other concerns you have.